### PR TITLE
Remove remaining traces of boost

### PR DESCRIPTION
**Pull request by fpsunflower**
## Description

Replace the last few uses of boost with robin-map (which is already a dependency for OIIO, although it is new for OSL itself) and functions in OIIO.

## Tests

Testsuite passes on my mac laptop. Will keep an eye on how the CI steps go.

## Checklist:

- [x] I have read the [contribution guidelines](../CONTRIBUTING.md).
- [x] I have updated the documentation, if applicable.
- [x] I have ensured that the change is tested somewhere in the testsuite (adding new test cases if necessary).
- [x] My code follows the prevailing code style of this project. If I haven't
  already run clang-format v17 before submitting, I definitely will look at
  the CI test that runs clang-format and fix anything that it highlights as
  being nonconforming.

### CODE REVIEW

File: src/doc/languagespec.tex [hunk 1]
```diff
@@ -182,7 +182,7 @@
 \smallskip
 \spc \includegraphics[width=0.85in]{Figures/CC-BY.png}
 \spc http://creativecommons.org/licenses/by/4.0/
- \bigskip
```
Comment (lgritz) [inline src/doc/languagespec.tex:185]: Can you please revert languagespec.tex? It's just making whitespace changes all throughout.

File: src/liboslexec/CMakeLists.txt [hunk 2]
```diff
@@ -62,13 +62,13 @@ set ( liboslexec_target_srcs
    ../liboslnoise/wide/wide_gabor_hybrid_disabled
    ../liboslnoise/wide/wide_gabor_hybrid_enabled
    ../liboslnoise/wide/wide_gabor_isotropic_disabled
-   ../liboslnoise/wide/wide_gabor_isotropic_enabled
```
Comment (lgritz) [inline src/liboslexec/CMakeLists.txt:65]: Can you revert all the whitespace changes here? I don't mind fixing that eventually, but I'd rather do it as a separate PR and do the git blame trick since nothing really changed.

**Comment by lgritz** [comment]
I would recommend copying OIIO's FindRobinmap.cmake and the parts of externalpackages.cmake that deal with robinmap, just exactly copy all of that because it works.

File: src/doc/languagespec.tex (hunk 1 above)
Comment (fpsunflower) [reply src/doc/languagespec.tex:185]: Will do - it looks like my editor is trimming whitespace on the whole file instead of on edited lines.

File: src/liboslexec/CMakeLists.txt (hunk 2 above)
Comment (fpsunflower) [reply src/liboslexec/CMakeLists.txt]: Fixed.

**Comment by fpsunflower** [comment]
> I would recommend copying OIIO's FindRobinmap.cmake and the parts of externalpackages.cmake that deal with robinmap, just exactly copy all of that because it works.

I gave a quick try to `FetchContent` since I was reading about it earlier and this is a particularly "easy" case with a header only dependency that has a modern cmake build system with exported targets. It seems to work as advertised, and will automatically do a `find_package` first to check if you already have the library on your system before executing `FetchContent`. 

It doesn't seem to work properly on the CI machines though -- and the error message is not helpful at all. Is there a trick to accessing the internet from the CI machines perhaps?

Of course I'm happy to clone the existing setup from OIIO if this fails, but the amount of boilerplate is definitely reduced with this approach.

**Comment by lgritz** [comment]
There is no trick, the CI runners have direct internet access.

**Comment by lgritz** [comment]
I would recommend just copying the old way OIIO did it, it has worked perfectly for a long time.

When I finish perfecting the new OIIO build system, we'll port it over here, too.

At that point, it will be easy to experiment with FetchContent if we want, and if that works any better or simpler instead of my build_dependency_with_cmake approach, I'm happy to switch.

**Comment by fpsunflower** [comment]
Will do.

**Comment by fpsunflower** [comment]
I believe this is working now. There are still some failures on OSX but they appear to be unrelated.

**Comment by fpsunflower** [comment]
I think partio was broken because it was previously pulling in zlib implictly via boost. It seems a bit weird that the cmake setup for partio doesn't handle this for us, but perhaps it isn't fully correct on their end?

Adding zlib to the target_link_libraries fixes it locally for me (was building without partio previously).

**Comment by lgritz** [comment]
> I believe this is working now. There are still some failures on OSX but they appear to be unrelated.

That's unrelated. I expected it to be fixed after the OIIO release a couple days ago. I'll need to look into it separately, but it's definitely not at all related to your PR.

**Review by lgritz — APPROVED** [review]
LGTM!
